Build & generator cleanup: drop Qt5 wrapper fallbacks, switch CI to Qt 6.10, normalize macOS includes, add qmake toggles, and regenerate wrappers in CI

hjmjohnson opened the pull request:
This PR modernizes PythonQt’s wrapper generation & build pipeline and aligns CI with Qt 6.10. It removes legacy, checked-in Qt5 wrapper directories, relies on explicit/generated paths at build time, normalizes macOS includes to bare Qt headers, adds qmake toggles for targeted builds, and updates CI to first generate wrappers then build without the generator. It also excludes a new `std::string_view` operator introduced in Qt 6.10 that is not yet supported by the generator.

### Summary

* Add `CONFIG+=generator_only` (build just the wrapper generator).
* Add `CONFIG+=exclude_generator` (build everything except the generator).
* **CI**
* Switch "latest" pipeline from **Qt 6.9 -> Qt 6.10** (workaround missing `Updates.xml.sha256` for 6.9.3).
* Restructure jobs to:

  1. build generator,
  2. run `pythonqt_generator`,
  3. build without the generator and run tests.
* On macOS, generate wrappers for **both Debug and Release**.
* **Repository housekeeping**

* Remove stale example wrapper trees:
  * `generated_cpp_50`, `generated_cpp_53`, `generated_cpp_54`, `generated_cpp_56`, `generated_cpp_511`
  * Keep `generated_cpp_515` as the canonical example.

### How to build now

Preferred: generate wrappers into `../generated_cpp` (relative to the build) **or** point qmake at an absolute path.

```bash
# Option A: generate to ../generated_cpp
qmake CONFIG+=generator_only
make -C generator
./generator/pythonqt_generator  # emits to ../generated_cpp by default

# Option B: use a custom location
qmake PYTHONQT_GENERATED_PATH=/abs/path/to/generated_cpp
```

Then build without the generator:

```bash
qmake CONFIG+=exclude_generator
make -j$(nproc)
```

If wrappers are missing, the build will fail early with a message listing searched locations and how to generate them.

### Notes

* **Removed fallbacks:** Projects that relied on the repository’s Qt5 wrapper directories must re-generate wrappers for their Qt version.
* **Qt 6.10:** CI now targets 6.10. Local builds with older Qt versions should still work if wrappers are generated for that version.
* **macOS:** Header normalization only affects generated include lines; no behavior change at runtime.
* **Temporarily excluded API:** The Qt 6.10 `std::string_view` operator is skipped in the typesystem until binding support is implemented.

jcfr commented:
@hjmjohnson 

Following integration of #308 , I rebased and started to improve your topic. Considering I do not have the credentials to update your fork, here is the updated topic:

https://github.com/MeVisLab/pythonqt/compare/master...jcfr:MeVisLab-pythonqt:remove-old-generated-code?compare=0

hjmjohnson commented:
@jcfr I've added you as a collaborator on my fork.  Thanks for the vastly improved suggestions.

hjmjohnson commented:
FYI: I'm wondering if the CI build resources should continue to be spent on all of these.

## Qt 5 End-of-Life / End-of-Support Overview

| Qt 5 Version / Branch | EOL / End of Standard Support | Notes / Extended Support |
|------------------------|-------------------------------|---------------------------|
| Qt 5.15 (LTS)          | 26 May 2025                  | After that, only via commercial/extended security maintenance. |
| Qt 5.14                | ~ 12 December 2020           | Dropped from standard support well before 5.15’s EOL. |
| Qt 5.13                | ~ 19 June 2020               | Already EOL well before the later 5.x versions. |
| Qt 5.12 (LTS)          | ~ 5 December 2021            | Last “open LTS” for much of its life; standard support ended earlier. |
| Qt 5.9                 | ~ 31 May 2020                | Non-LTS branch; short support window. |
| Qt 5.6 (LTS)           | ~ 16 March 2019              | Very old LTS, support ended long ago. |

hjmjohnson commented:
@jcfr. FYI:  The scope of fixing the CI environment for generating the required files dynamically is beyond what I will be able to contribute in the short run.  

I've converted this PR to a draft and simplified it to attack one issue at a time.

hjmjohnson commented:
@jcfr You are awesome!  Thank you for taking a closer look at the CI and improving its robustness.

hjmjohnson commented:
@jcfr  Looks really close to working.   I'm surprised that the debug variant passes, but the release variant fails with a missing header:  https://github.com/MeVisLab/pythonqt/actions/runs/18065296643/job/51407182062?pr=313#step:9:124 error.

hjmjohnson commented:
@jcfr You may already know of this tool, but I thought I'd put it here for documentation purposes:

https://ddalcino.github.io/aqt-list-server/

jcfr commented:
> Build / macOS (latest, 3.9, 5.9.*, release)

```
/Users/runner/work/pythonqt/Qt/5.9.9/clang_64/bin/moc -DPYTHONQT_CATCH_ALL_EXCEPTIONS -DPYTHONQT_EXPORTS -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_CORE_LIB --include ./moc_predefs.h -I/Users/runner/work/pythonqt/Qt/5.9.9/clang_64/mkspecs/macx-clang -I/Users/runner/work/pythonqt/pythonqt/src -I/Users/runner/work/pythonqt/pythonqt/src -I/Users/runner/work/pythonqt/pythonqt/src -I/Library/Frameworks/Python.framework/Versions/3.9/include/python3.9 -I/Users/runner/work/pythonqt/Qt/5.9.9/clang_64/lib/QtWidgets.framework/Headers -I/Users/runner/work/pythonqt/Qt/5.9.9/clang_64/lib/QtGui.framework/Headers -I/Users/runner/work/pythonqt/Qt/5.9.9/clang_64/lib/QtCore.framework/Headers/5.9.9 -I/Users/runner/work/pythonqt/Qt/5.9.9/clang_64/lib/QtCore.framework/Headers/5.9.9/QtCore -I/Users/runner/work/pythonqt/Qt/5.9.9/clang_64/lib/QtCore.framework/Headers -I/Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.5.sdk/usr/include/c++/v1 -I/Applications/Xcode_16.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/17/include -I/Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.5.sdk/usr/include -I/Applications/Xcode_16.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include -F/Users/runner/work/pythonqt/Qt/5.9.9/clang_64/lib ../generated_cpp/com_trolltech_qt_gui_builtin/com_trolltech_qt_gui_builtin0.h -o moc_com_trolltech_qt_gui_builtin0.cpp
In file included from ../generated_cpp/com_trolltech_qt_core_builtin/com_trolltech_qt_core_builtin0.cpp:1:
../generated_cpp/com_trolltech_qt_core_builtin/com_trolltech_qt_core_builtin0.h:1:10: fatal error: '../lib/QtCore.framework/Headers/qbitarray.h' file not found
    1 | #include <../lib/QtCore.framework/Headers/qbitarray.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../generated_cpp/com_trolltech_qt_gui_builtin/com_trolltech_qt_gui_builtin0.cpp:1:
../generated_cpp/com_trolltech_qt_gui_builtin/com_trolltech_qt_gui_builtin0.h:1:10: fatal error: '../lib/QtCore.framework/Headers/qbytearray.h' file not found
    1 | #include <../lib/QtCore.framework/Headers/qbytearray.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

jcfr commented:
> re: error downloading Qt 6.9.3

I am able to reproduce locally:

```
$ uvx --from aqtinstall aqt install-qt linux desktop 6.9.* linux_gcc_64 --autodesktop --outputdir /tmp/Qt --modules qt5compat qtscxml qtpositioning qtwebchannel qtmultimedia qtwebengine
INFO    : aqtinstall(aqt) v3.3.0 on Python 3.10.12 [CPython GCC 11.4.0]
INFO    : Resolved spec '6.9.*' to 6.9.3
WARNING : Failed to download checksum for the file 'online/qtsdkrepository/linux_x64/desktop/qt6_693/qt6_693/Updates.xml'. This may happen on unofficial mirrors.
ERROR   : The packages ['qtwebengine'] were not found while parsing XML of package information!
==============================Suggested follow-up:==============================
```

jcfr commented:
When dowloading `6.10.*`, the following checksum is looked up for [qt6_6100/qt6_6100/Updates.xml](https://download.qt.io/online/qtsdkrepository/linux_x64/desktop/qt6_6100/qt6_6100/Updates.xml):

https://download.qt.io/online/qtsdkrepository/linux_x64/desktop/qt6_6100/qt6_6100/Updates.xml.sha256

where when dowloading `6.9.*`, the following checksum is looked up for [qt6_693/qt6_693/Updates.xml](https://download.qt.io/online/qtsdkrepository/linux_x64/desktop/qt6_693/qt6_693/Updates.xml):

https://download.qt.io/online/qtsdkrepository/linux_x64/desktop/qt6_693/qt6_693/Updates.xml.sha256

jcfr commented:
.. but the file https://download.qt.io/online/qtsdkrepository/linux_x64/desktop/qt6_693/qt6_693/Updates.xml.sha256 does not exist anymore

<img width="1622" height="287" alt="image" src="https://github.com/user-attachments/assets/0c437f2e-5680-495b-9e12-d380ac0137a6" />

jcfr commented:
Looking into this error reported after updating CI to build against 6.10:

```
..\generated_cpp\com_trolltech_qt_core_builtin\com_trolltech_qt_core_builtin0.h -o moc_com_trolltech_qt_core_builtin0.cpp
..\generated_cpp\com_trolltech_qt_core_builtin\com_trolltech_qt_core_builtin0.h(154:1): error: Not a signal or slot declaration
```

.. inspecting `com_trolltech_qt_core_builtin0.h` seems to indicate there are additional spaces: 

```
   [...]
   QByteArray  static_QByteArray_number(ulong  arg__1, int  base = 10);
   QByteArray  operator std :: string_view(QByteArray* theWrappedObject) const;
   bool  __ne__(QByteArray* theWrappedObject, const QByteArray&  rhs);
   [...]
```

It should likely be `std::string_view` instead of std :: string_view
